src/mds/MDCache.cc

```diff
@@ -13623,31 +13623,37 @@ void MDCache::dispatch_quiesce_inode(const MDRequestRef& mdr)
 
     lov.add_xlock(&in->quiescelock); /* !! */
 
-    lov.add_rdlock(&in->policylock); /* for the quiesce_block xattr test */
-    if (splitauth) {
-      // xlock the file to let the Fb clients stay with buffered writes.
-      // While this will unnecesarily revoke rd caps, it's not as
-      // big of an overhead compared to having the Fb clients flush
-      // their buffers, which evidently can lead to the quiesce timeout
-      // We'll drop the lock after all clients conform to this request
-      // so the file will be still readable during the quiesce after
-      // the interested clients receive their Fr back
-      //
-      // NB: this will also wrlock the versionlock
-      lov.add_xlock(&in->filelock);
+    if (in->is_auth()) {
+      lov.add_rdlock(&in->policylock); /* for the quiesce_block xattr test */
```

Comment:
replica mds needs this lock too?

Reply:
I don't think so, since it's a SimpleLock, so it's mirrored

Reply:
hmm... you are right; but this makes me wonder if we have an issue here with a window where the auth and the replica may see different values of the `quiesce_block` xattr for the same quiesce.

Reply:
The only way I can think of that will prevent that is that we don't release the policylock, at least not until the quiesce is complete on all ranks

Reply:
OK I don't think we have the time for this. We'll just have to mention in the docs (are there docs about it?) that the quiesce.block shouldn't be set during quiesces, for now.

Though before we add it to the docs, I'd still like to rename it to `quiesce.skip` cause `block` still throws me off, TBH.

Reply:
Alternatively, we could remove this feature until we re-introduce it consistently.

Reply:
Or, how about this: it appears that the `policylock` doesn't help us with the race of a quiesce and setting of this flag. So, to be safe, the user would need to change this flag and then re-issue a quiesce. If that's the case, we may as well not even bother taking the `policylock` :D

Reply:
If the lock is not in `sync` state, you cannot safely read it.

Please pull the commit out. I have discovered several bugs surrounding the policylock which I'll post in a separate PR.

Reply:
Ack!
We have to figure out a new way of setting the quiesce skip flag to be consistent across the cluster with a racing quiesce. Having a separate PR for that makes a lot of sense.

The current solution seems good enough for the first version of the quiesce, it will work well if we need to enable it in field as it will most probably mean some more or less static config under support supervision